invenio_rdm_records/services/permissions.py

```diff
@@ -112,6 +112,8 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
             else_=can_read,
         )
     ]
+    can_read_deleted_files = can_read_deleted
```

Comment:
IMO it would be safer if you set `can_read_deleted + can_read_files`. This will be additional layer of security in case of some instance having a need of hiding the content of the files also from the moderator

Reply:
after going back and forth between here and records resources I understand that it is being checked separately and adding those generator lists will actually not change much, you can leave it as it and I will take into account when I refactor the deletion of the records.
The reason for all this thinking is that I worry if we keep can_read_deleted_files separate from can_read_files leaves a door for mistakes if someone will overwrite service methods in their instances